src/components/form.tsx

```diff
@@ -2,9 +2,10 @@ import * as React from 'react';
 import { Link } from 'react-router-dom';
 
 import { default as Button, ButtonStyle } from './button';
+import {TranslatedText} from '../utils/i18n';
 
 
-export type FormValidationResponse = { [key: string]: any[] | string }; // can be any[] if returned from I18n.format
+export type FormValidationResponse = { [key: string]: TranslatedText | FormValidationResponse };
```

Comment:
why can FormValidationResponse = FormValidationResponse?

Reply:
Sometimes they are nested (for example the material spec sheet link pairs that Daniel is working on for the contribute portal.